### PR TITLE
more performant changing of reasons

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
pref refactoring

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

File: test/__snapshots__/StatsTestCases.test.js.snap [hunk 1]
```diff
@@ -2063,52 +2063,49 @@ Child
...
 exports[`StatsTestCases should print correct stats for side-effects-issue-7428 1`] = `
-"Hash: 9d82ff1a0201e9c2ffa4
+"Hash: 3820b12fcca2168af8bc
 Time: Xms
 Built at: Thu Jan 01 1970 00:00:00 GMT
   Asset       Size  Chunks             Chunk Names
-   0.js  481 bytes       0  [emitted]  
-main.js   9.31 KiB       1  [emitted]  main
+   0.js  599 bytes       0  [emitted]  
+main.js   9.54 KiB       1  [emitted]  main
```
Comment (ooflorent) [inline test/__snapshots__/StatsTestCases.test.js.snap]: This is concerning. Why are there changes here?

Comment (montogeek) [reply test/__snapshots__/StatsTestCases.test.js.snap]: He updated snapshots to make tests pass

Comment (ooflorent) [reply test/__snapshots__/StatsTestCases.test.js.snap]: Yes but the modification shouldn’t change this.

Comment (sokra) [reply test/__snapshots__/StatsTestCases.test.js.snap]: Yep, something is wrong

**Pull request merged**